aten/src/ATen/core/ivalue_inl.h

```diff
@@ -412,10 +413,16 @@ struct C10_EXPORT ivalue::Future : c10::intrusive_ptr_target {
     if (completed_) {
       // This should be rare and shouldn't cause log spew. Its important to
       // log errors and thats why we have this log here.
-      LOG(INFO)
-          << "Skipping setting following error on the Future since "
-          << "it is already marked completed (this is not neccessarily an error): "
-          << tryRetrieveErrorMessageInternal(eptr);
+      std::string msg = fmt::format(
+          "Skipping setting following error on the Future since "
+          "it is already marked completed (this is not neccessarily "
+          "an error):\n {}", tryRetrieveErrorMessageInternal(eptr));
+      if (eptr_) {
+        msg += fmt::format(
+            ", \nOriginal exception:\n {}",
+            tryRetrieveErrorMessageInternal(eptr_));
+      }
```

Comment:
nit: If it was completed without an error, should we log that as well (and possibly the IValue it was completed with)? We should usually have an error so it might be good if that happens to add it as a `LOG(WARNING)`.